crates/node/p2p/src/gossip/driver.rs

```diff
@@ -198,36 +171,20 @@ impl GossipDriver {
 
     /// Dials the given [`Multiaddr`].
     pub fn dial_multiaddr(&mut self, addr: Multiaddr) {
-        let Some(peer_id) = addr
-            .iter()
-            .find_map(|p| if let Protocol::P2p(peer_id) = p { Some(peer_id) } else { None })
-        else {
-            warn!(target: "gossip", peer=?addr, "Failed to extract peer id from multiaddr");
-            return;
-        };
-
-        if self.swarm.connected_peers().any(|p| p == &peer_id) {
-            debug!(target: "gossip", peer=?addr, "Already connected to peer, not dialing");
-            return;
-        }
-
-        if self.current_dials.contains(&peer_id) {
-            debug!(target: "gossip", peer=?addr, "Already dialing peer, not dialing");
-            return;
-        }
-
-        if self.dial_threshold_reached(&addr) {
-            debug!(target: "gossip", peer=?addr, "Dial threshold reached, not dialing");
+        // Check if we're allowed to dial the address.
+        if !self.connection_gate.can_dial(&addr) {
+            warn!(target: "gossip", "[GATE] unable to dial peer");
```

Comment:
tiny nit - rm `[GATE]`
```suggestion
            warn!(target: "gossip", "unable to dial peer");
```

Reply:
okie, I like these flags so it's ez to see in logs but fine to rm